ci/ci/build.py

```diff
@@ -88,22 +88,22 @@ def build(self, batch, code, scope):
             if step.scopes is None or scope in step.scopes:
                 step.build(batch, code, scope)
 
-        parents = set()
-        for step in self.steps:
-            parents.update(step.wrapped_job())
-        parents = list(parents)
-
         if scope == 'dev':
             return
 
-        sink = batch.create_job('ubuntu:18.04',
-                                command=['/bin/true'],
-                                attributes={'name': 'sink'},
-                                parents=parents)
+        step_to_parent_steps = {}
```

Comment:
use a [`defaultdict`](https://docs.python.org/3.3/library/collections.html#collections.defaultdict), it'll make the code clearer.

Also I think this would break on line 103 for steps with no dependencies (i.e. `default_ns`).